src/sign-transaction/SignTransactionWithPin.js

```diff
@@ -0,0 +1,78 @@
+/** Handles a sign-transaction request for keys with encryption type LOW.
+ *  Calls this.fire('result', [result]) when done or this.fire('error', [error]) to return with an error.
+ */
+class SignTransactionWithPin extends Nimiq.Observable {
+
+    /** @param {TransactionRequest} txRequest */
+    constructor(txRequest) {
+        super();
+
+        this._txRequest = txRequest;
+
+        // construct UI
+        const rootElement = document.getElementById('app');
+
+        if (!rootElement) {
+            throw new InvalidDOMError();
+        }
+
+        const $button = rootElement.querySelector('#transaction-data button');
+        const $enterPin = rootElement.querySelector('#enter-pin');
+
+        if (!$button || !$enterPin) {
+            throw new InvalidDOMError();
+        }
+
+        $button.addEventListener('click', () => location.hash = 'enter-pin');
+
+        this.$pinInput = new PinInput();
+
+        $enterPin.appendChild(this.$pinInput.getElement());
+
+        this.$pinInput.open();
+
+        this.$pinInput.on('pin-entered', this.handlePinInput.bind(this));
+
+        // go to start page
+        location.hash = 'transaction-data'
```

Comment:
Maybe these should also be collected as constants in an object:

SignTransactionsWithPin.Pages = {
    TRANSACTION_DATA: 'transaction-data',
    ENTER_PIN: 'enter-pin'
};